File: src/types/license.ts
export type LicenseType = 'basic' | 'pro' | 'enterprise';

export interface LicenseStatus {
  is_activated: boolean;
  is_valid: boolean;
  expires_at?: number | null;
  email?: string | null;
  days_remaining?: number | null;
  license_type?: string | null;
  error_message?: string | null;
}

export interface LicenseActivationRequest {
  key: string;
  email: string;
}

export interface LicenseValidationRequest {
  key: string;
  email: string;
  machine_fingerprint: string;
}

export interface LicenseValidationResponse {
  valid: boolean;
  expires_at?: number;
  user_email: string;
  license_type: string;
  error?: string;
}

File: src-tauri/src/models/license.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LicenseKey {
    pub key_hash: String,
    pub email: String,
    pub machine_fingerprint: String,
    pub activated_at: i64,
    pub expires_at: Option<i64>,
    pub is_active: bool,
    pub license_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LicenseValidationRequest {
    pub key: String,
    pub email: String,
    pub machine_fingerprint: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LicenseValidationResponse {
    pub valid: bool,
    pub expires_at: Option<i64>,
    pub user_email: String,
    pub license_type: String,
    pub error: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LicenseStatus {
    pub is_activated: bool,
    pub is_valid: bool,
    pub expires_at: Option<i64>,
    pub email: Option<String>,
    pub days_remaining: Option<i64>,
    pub license_type: Option<String>,
    pub error_message: Option<String>,
}
